lib/scsi_defs.h

```diff
@@ -874,6 +884,56 @@ struct UnmapBlockDescriptor {
 } ABSL_ATTRIBUTE_PACKED;
 static_assert(sizeof(UnmapBlockDescriptor) == 16);
 
+// SCSI Reference Manual Table 483
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct SupportedVitalProductData {
+  PeripheralQualifier peripheral_qualifier : 3;
+  PeripheralDeviceType peripheral_device_type : 5;
+  PageCode page_code : 8;
+  uint8_t _reserved : 8;
+  uint8_t page_length : 8;
+  // PageCode supported_page_list[256];
+} ABSL_ATTRIBUTE_PACKED;
+static_assert(sizeof(SupportedVitalProductData) == 4);
+
+// SCSI Reference Manual Table 484
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct UnitSerialNumber {
+  PeripheralQualifier peripheral_qualifier : 3;
+  PeripheralDeviceType peripheral_device_type : 5;
+  PageCode page_code : 8;
+  uint8_t _reserved : 8;
+  uint8_t page_length : 8;
```

Comment:
I suggest creating an enum for PageLength and include the page length of all the structures with fixed page lengths

Reply:
this will be a change when we merge all inquiry things together since you've done this already